src/cascadia/TerminalSettingsModel/Theme.cpp

```diff
@@ -0,0 +1,331 @@
+
+#define THEME_OBJECT_CONVERTER(nameSpace, name, macro)                                         \
+    template<>                                                                                 \
+    struct ::Microsoft::Terminal::Settings::Model::JsonUtils::ConversionTrait<nameSpace::name> \
+    {                                                                                          \
+        nameSpace::name FromJson(const Json::Value& json)                                      \
+        {                                                                                      \
+            if (json == Json::Value::null)                                                     \
+                return nullptr;                                                                \
+            auto result = winrt::make_self<nameSpace::implementation::name>();                 \
+            macro(THEME_SETTINGS_FROM_JSON);                                                   \
+            return *result;                                                                    \
+        }                                                                                      \
+                                                                                               \
+        bool CanConvert(const Json::Value& json)                                               \
+        {                                                                                      \
+            return json.isObject();                                                            \
+        }                                                                                      \
+                                                                                               \
+        Json::Value ToJson(const nameSpace::name& val)                                         \
+        {                                                                                      \
+            if (val == nullptr)                                                                \
+                return Json::Value::null;                                                      \
+            Json::Value json{ Json::ValueType::objectValue };                                  \
+            macro(THEME_SETTINGS_TO_JSON);                                                     \
+            return json;                                                                       \
+        }                                                                                      \
+                                                                                               \
+        std::string TypeDescription() const                                                    \
+        {                                                                                      \
+            return "name (You should never see this)";                                         \

```

Comment:
lol this will actually say the word "name" since it's not in a macro-stringify expression